### ISSUE TITLE
Time field in Point is restricted to Long

### ISSUE THREAD

**Issue by J0s3f** [influxdata/influxdb-client-java#85]
It would make sense to make it possible to use something bigger than long for time in the Point class.
We use nanosecond precision for all data in an project, and Long just isn't suitable.
In the "old" client there is a [similar bug](https://github.com/influxdata/influxdb-java/issues/267) and I tried to fix it by allowing Number for all timestamps.

The proper solution IMO would be to have points with Instant as time. Instant is the semantically correct class to use and converting an instant into nanoseconds for the line protocol works by simply converting the epoch seconds to string and appending the nanoseconds as string.

**Comment by bednar** [comment]
Hi @J0s3f,

thanks for using our library.

The [Point](https://github.com/influxdata/influxdb-client-java/blob/b3ce37dfbeba5d7a77ff14f4d03ae3b0414582d9/client/src/main/java/com/influxdb/client/write/Point.java#L51) supports Instant. You could use it this way:

```java
Instant instant = Instant.parse("2020-02-15T10:20:30.123456789Z");

Point point = Point.measurement("h2o")
                .addTag("location", "europe")
                .addField("level", 2)
                .time(instant, WritePrecision.NS);

System.out.println("point.toLineProtocol() = " + point.toLineProtocol());
```
```
> h2o,location=europe level=2i 1581762030123456789
```

The `long` has space to store time up to year 2262. It is not enough?

Regards

**Comment by bednar** [comment]
Hi @J0s3f,

The issue is fixed in [1.6.0 milestone](https://github.com/influxdata/influxdb-client-java/milestone/9). 

If you would like to use a preview version use our snapshot build from repository: https://oss.sonatype.org/content/repositories/snapshots/

Regards

**Comment by J0s3f** [comment]
Thanks, seems to work!
